docs/source/usage.rst

```diff
@@ -0,0 +1,230 @@
+
+Attribute validation includes checks against illegal attribute value
+assignments. This means that value assignments for attributes which are
+excluded for the model are treated with
+:class:`~eav.exceptions.IllegalAssignmentException`. For example (extending
+previous one):
+
+.. code-block:: python
+
+    some_model.eav.beard = True
+    some_model.save()
+
+will throw an exception.
+
+Creating Attributes
+-------------------
+
+Once your models are registered, you can starting creating attributes for
+them. Two most important attribute attributes (huh?) are *slug* and
+*datatype*. *slug* is a unique identifier and must be a valid Python
+variable name, as it's used to access values for that attribute from
+:class:`~eav.models.Entity` helper:
+
+.. code-block:: python
+
+    Attribute.objects.create(slug='color', datatype=Attribute.TYPE_TEXT)
+    flower.eav.color = 'red'
+
+    # Alternatively, assuming you're using default EntityManager:
+    Flower.objects.create(name='rose', eav__color='red')
+
+*datatype* determines type of attribute (and by extension type of value
+stored in :class:`~eav.models.Value`). Available choices are:
+
+========= ==================
+  Type    Attribute Constant
+========= ==================
+*int*     ``TYPE_INT``
+*float*   ``TYPE_FLOAT``
+*text*    ``TYPE_TEXT``
+*date*    ``TYPE_DATE``
+*bool*    ``TYPE_BOOLEAN``
+*object*  ``TYPE_OBJECT``
+*enum*    ``TYPE_ENUM``
+========= ==================
+
+If you want to create an attribute with data-type *enum*, you need to provide
+it with ``enum_group``:
+
+.. code-block:: python
+
+    true = EnumValue.objects.create(value='Yes')
+    false = EnumValue.objects.create(value='No')
+    bool_group = EnumGroup.objects.create(name='Yes / No')
+    bool_group.enums.add(true, false)
+
+    Attribute.objects.create(
+        name='hungry?',
+        datatype=Attribute.TYPE_ENUM,
+        enum_group=bool_group
+    )
+    # = <Attribute: hungry? (Multiple Choice)>
+
+Finally, attribute type *object* allows to relate Django model instances
+via generic foreign keys:
+
+.. code-block:: python
+
+    Attribute.objects.create(name='Supplier', datatype=Attribute.TYPE_OBJECT)
+
+    steve = Supplier.objects.create(name='Steve')
+    cog = Part.objects.create(name='Cog', eav__supplier=steve)
+
+    cog.eav.supplier
+    # = <Supplier: Steve (1)>
+
+Filtering By Attributes
+-----------------------
+
+Once you've created your attributes and values for them, you can use them
+to filter Django models. Django EAV is using the same notation as Django's
+foreign-keys:
+
+.. code-block:: python
+
+    Part.objects.filter(eav__weight=10)
+    Part.objects.filter(eav__weight__gt=10)
+    Part.objects.filter(eav__code__startswith='A')
+
+    # Of course, you can mix them with regular queries:
+    Part.objects.filter(name='Cog', eav__height=7.8)
+
+    # Querying enums looks as follows:
+    yes = EnumValue.objects.get(name='Yes')
+    Part.objects.filter(eav__is_available=yes)
```

Comment:
It's not possible to go deeper, `eav__is_available__name='yes'`, isn't it?

Reply:
Not sure. I'll look into that. If not, that's a good feature-request.